### PR TITLE
Feature/dp setup do

### PR DESCRIPTION
**Pull request by DenisPopov15**

### CODE REVIEW

File: bin/do/setup [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+worker_processes 4
+user "rails"
+working_directory "/home/rails/rails_project"
+pid "/var/run/unicorn.pid"
+stderr_path "/var/log/unicorn/unicorn.log"
+stdout_path "/var/log/unicorn/unicorn.log"
+EOT
+
+SECRET_KEY_BASE=$(rake -f /home/rails/rails_project/Rakefile secret)
+cat <<EOT >> /var/rails_vars
```
Comment (alexkravets) [inline bin/do/setup]: Let's put this file to /etc/default/rails_project.

File: bin/do/deploy [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/bin/sh
+
+if [ "$*" == "" ]; then
+  echo "Please provide host/api, see: https://cloud.digitalocean.com/droplets"
```
Comment (alexkravets) [inline bin/do/deploy]: host/ip

File: bin/do/setup [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+# Server's config.rb, it's not a rack's config.ru
+CONFIG_RB=/etc/unicorn.conf
+
+# Where to store PID, sh'ld be also set in server's config.rb, option "pid".
+PID=/var/run/unicorn.pid
+RAILS_ENV="production"
+UNICORN_OPTS="-D -c \$CONFIG_RB -E \$RAILS_ENV"
+
+PATH=/usr/local/rvm/rubies/$RUBY_VERSION/bin:/usr/local/sbin:/usr/bin:/bin:/sbin:/usr/local/rvm/bin:/usr/local/rvm/gems/$RUBY_VERSION@global/bin:/usr/local/rvm/gems/$RUBY_VERSION/bin/
+export GEM_HOME=/usr/local/rvm/gems/$RUBY_VERSION
```
Comment (alexkravets) [inline bin/do/setup]: Let's have $RUBY_VERSION defined in this file and use variable here and below. So if $RUBY_VERSION changes we just need to change it once.

File: bin/do/setup [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.0.list
+# ​Reload local package database:
+sudo apt-get update
+echo "Installing Mongo DB"
+sudo apt-get install -y mongodb-org
+export LC_ALL=C
+# Create `rails` mongodb user in venice db:
+mongo --eval "conn = new Mongo(); db = conn.getDB('venice'); db.createUser({user:'rails', pwd:'$MONGO_PASS', roles:['readWrite']})"
+
+adduser rails --gecos "First Last,RoomNumber,WorkPhone,HomePhone" --disabled-password
```
Comment (alexkravets) [inline bin/do/setup]: I think we should use --disable-login here, so rails user can't login to the system.

File: bin/do/setup [hunk 5]
```diff
@@ -0,0 +1,263 @@
+#!/bin/sh
+
+# On Server
+# -------------------------------------------------
+GIT_URL="$1"
+HOST="$2"
+MONGO_PASS=$(date +%s | sha256sum | base64 | head -c 32 ; echo)
+
+sudo apt-get update
```
Comment (alexkravets) [inline bin/do/setup]: Why are you using sudo here an below? You're logged in as root, so no need in `sudo` which is `super user do` = `root do`.

**Comment by DenisPopov15** [comment]
@alexkravets updated.

File: bin/do/setup [hunk 6]
```diff
@@ -0,0 +1,265 @@
...
+# Server's config.rb, it's not a rack's config.ru
+CONFIG_RB=/etc/unicorn.conf
+
+# Where to store PID, sh'ld be also set in server's config.rb, option "pid".
+PID=/var/run/unicorn.pid
+RAILS_ENV="production"
+UNICORN_OPTS="-D -c \$CONFIG_RB -E \$RAILS_ENV"
+
+RUBY_VERSION=$RUBY_VERSION
+PATH=/usr/local/rvm/rubies/$RUBY_VERSION/bin:/usr/local/sbin:/usr/bin:/bin:/sbin:/usr/local/rvm/bin:/usr/local/rvm/gems/$RUBY_VERSION@global/bin:/usr/local/rvm/gems/$RUBY_VERSION/bin/
```
Comment (alexkravets) [inline bin/do/setup]: Here you should escape $RUBY_VERSION now with \
Please test the script after changes.

File: bin/do/setup [hunk 7]
```diff
@@ -0,0 +1,265 @@
...
+apt-get -y install libgmp3-dev
+apt-get -y install imagemagick
+apt-get -y install git-core
+apt-get -y install nodejs
+
+## MongoDB 3
+# Add public key:
+apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv 7F0CEB10
+# Create a list file for MongoDB:
+echo "deb http://repo.mongodb.org/apt/ubuntu trusty/mongodb-org/3.0 multiverse" | sudo tee /etc/apt/sources.list.d/mongodb-org-3.0.list
```
Comment (alexkravets) [inline bin/do/setup]: Remove here sudo as well.

**Comment by alexkravets** [comment]
Please rename bin/do to bin/digital-ocean — as it may be not obvious what do stands for.

File: bin/do/setup (hunk 6 above)
Comment (DenisPopov15) [inline bin/do/setup]: @alexkravets Yes, I figure out this and fixed, but you already made comment.

**Comment by DenisPopov15** [comment]
@alexkravets tested, all works fine.

File: bin/digital-ocean/setup [hunk 8]
```diff
@@ -0,0 +1,266 @@
...
+
+    location / {
+            try_files \$uri/index.html \$uri.html \$uri @app;
+    }
+
+    location ~* ^.+\.(jpg|jpeg|gif|png|ico|zip|tgz|gz|rar|bz2|doc|xls|exe|pdf|ppt|txt|tar|mid|midi|wav|bmp|rtf|mp3|flv|mpeg|avi)\$ {
+                    try_files \$uri @app;
+            }
+
+     location @app {
```
Comment (alexkravets) [inline bin/digital-ocean/setup]: Remove one space here.

File: bin/digital-ocean/setup [hunk 9]
```diff
@@ -0,0 +1,266 @@
...
+# Server's config.rb, it's not a rack's config.ru
+CONFIG_RB=/etc/unicorn.conf
+
+# Where to store PID, sh'ld be also set in server's config.rb, option "pid".
+PID=/var/run/unicorn.pid
+RAILS_ENV="production"
+UNICORN_OPTS="-D -c \$CONFIG_RB -E \$RAILS_ENV"
+
+RUBY_VERSION=$RUBY_VERSION
+PATH=/usr/local/rvm/rubies/\$RUBY_VERSION/bin:/usr/local/sbin:/usr/bin:/bin:/sbin:/usr/local/rvm/bin:/usr/local/rvm/gems/\$RUBY_VERSION@global/bin:/usr/local/rvm/gems/\$RUBY_VERSION/bin/
```
Comment (alexkravets) [inline bin/digital-ocean/setup:137]: Let's try to replace PATH, GEM_HOME and GEM_PATH definition with with `rvm use $RUBY_VERSION` — I guess it does the same.

Comment (alexkravets) [inline bin/digital-ocean/setup:137]: And maybe `rvm rubygems current` after `rvm use $RUBY_VERSION`.

File: bin/digital-ocean/setup [hunk 10]
```diff
@@ -0,0 +1,266 @@
...
+        ;;
+  *)
+        log_action_msg "Usage: \$0 <start|stop|restart|force-reload|reload|force-stop|reopen-logs|status>" || true
+        exit 1
+        ;;
+esac
+EOT
+
+chmod +x /etc/init.d/unicorn
+update-rc.d unicorn defaults
```
Comment (alexkravets) [inline bin/digital-ocean/setup:260]: What this is for? Please add a comment.

File: bin/digital-ocean/setup [hunk 11]
```diff
@@ -0,0 +1,266 @@
...
+RAILS_ENV="production"
+UNICORN_OPTS="-D -c \$CONFIG_RB -E \$RAILS_ENV"
+
+RUBY_VERSION=$RUBY_VERSION
+PATH=/usr/local/rvm/rubies/\$RUBY_VERSION/bin:/usr/local/sbin:/usr/bin:/bin:/sbin:/usr/local/rvm/bin:/usr/local/rvm/gems/\$RUBY_VERSION@global/bin:/usr/local/rvm/gems/\$RUBY_VERSION/bin/
+export GEM_HOME=/usr/local/rvm/gems/\$RUBY_VERSION
+export GEM_PATH=/usr/local/rvm/gems/\$RUBY_VERSION:/usr/local/rvm/gems/\$RUBY_VERSION@global
+DAEMON="\$APP_ROOT/bin/bundle exec unicorn"
+
+. /etc/default/rails_project
```
Comment (alexkravets) [inline bin/digital-ocean/setup:142]: I guess this should be replaced with `source /etc/default/rails_project` —as with rvm above.

File: bin/digital-ocean/setup [hunk 12]
```diff
@@ -0,0 +1,266 @@
...
+EOT
+
+chmod +x /etc/init.d/unicorn
+update-rc.d unicorn defaults
+service unicorn restart
+service nginx restart
+# --------------------------------------------------------------
+
+# Setup DB
+. /etc/default/unicorn
```
Comment (alexkravets) [inline bin/digital-ocean/setup]: Use `source` here instead of `.` as well.

File: bin/digital-ocean/setup (hunk 10 above)
Comment (alexkravets) [inline bin/digital-ocean/setup:260]: Ha I remember, this adds `unicorn` to `rc.d` links.

File: bin/digital-ocean/setup [hunk 13]
```diff
@@ -0,0 +1,266 @@
...
+  *)
+        log_action_msg "Usage: \$0 <start|stop|restart|force-reload|reload|force-stop|reopen-logs|status>" || true
+        exit 1
+        ;;
+esac
+EOT
+
+chmod +x /etc/init.d/unicorn
+update-rc.d unicorn defaults
+service unicorn restart
```
Comment (alexkravets) [inline bin/digital-ocean/setup]: Let's do `service unicorn start`.

File: bin/digital-ocean/setup (hunk 11 above)
Comment (DenisPopov15) [inline bin/digital-ocean/setup:142]: @alexkravets with `source...` it's not working. I tried it before use `. /etc....` .When this file calling fron `/etc/init.d/unicorn` like `. /etc/default/unicorn` `source..` some reason brake thing.

File: bin/digital-ocean/setup (hunk 9 above)
Comment (DenisPopov15) [inline bin/digital-ocean/setup:137]: @alexkravets great Idea, and in generally it must works, but I getting `Warning: PATH set to RVM ruby but GEM_HOME and/or GEM_PATH not set, see:
    https://github.com/wayneeseguin/rvm/issues/3212`. I researched this issue, tried different solutions, but somewhy it's don't solved problem for this config unicorn file. If run `rvm use..` not in config unicorn it's set this variables fine.

**Comment by DenisPopov15** [comment]
@alexkravets updated. About  PATH, GEM_HOME and GEM_PATH see my comment above.

**Pull request merged**